src/hooks/use-read-resource.ts

```diff
@@ -138,17 +138,13 @@ export function useReadResource<T>(resourceLike: ResourceLike<T>, options: UseRe
     if (cachedState) {
       setResourceState(cachedState);
       setLoading(false);
+
       return;
-    } else {
-      setResourceState(undefined);
-      setLoading(true);
     }
+    setResourceState(undefined);
+    setLoading(true);
 
     resource.get({ headers: options.initialGetRequestHeaders })
-      .then(newState => {
```

Comment:
These can be removed because `resource.get` emits the `update` event internally when it's updating the Ketting cache and that event is handled by the `onUpdate` above, right? So these two set state functions are redundant then?

Reply:
Correct! The update event was added after, but didn't realize at the time this one was now redundant

Reply:
Dope